app/forms/date_form.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+# Form object for handling date input
+class DateForm < ApplicationForm
+  attribute :year, :integer
+  validates :year,
+            numericality: { greater_than_or_equal_to: 1000, less_than_or_equal_to: Time.zone.now.year, allow_nil: true }
```

Comment:
Should we consider constantizing this to give it a name?

Reply:
Can you say more? I'm not sure what you're suggesting.

Reply:
Oh, my bad. I meant:

```
class DateForm < ApplicationForm
  EARLIEST_YEAR = 1000
  # ...
  validates :year,
            numericality: { greater_than_or_equal_to: EARLIEST_YEAR, less_than_or_equal_to: Time.zone.now.year, allow_nil: true }
```

Reply:
I'm not a big fan of constantizing things that only appear once. To me it feels like the indirection adds cognitive overhead. (But I don't feel that strongly about this.)